File: frontend/src/services/review.ts
import apiClient from './api'

export interface ReviewResponse {
  id: number
  venue_id: number
  user_id: number
  rating: number
  comment: string | null
  created_at: string
  user_name: string | null
  venue_name: string | null
}

export interface ReviewCreate {
  venue_id: number
  rating: number
  comment?: string | null
}

export interface VenueRatingSummary {
  venue_id: number
  average_rating: number
  total_reviews: number
}

export const reviewService = {
  // دریافت نظرات یک سالن
  getByVenue: async (venueId: number, limit = 50, offset = 0): Promise<ReviewResponse[]> => {
    const response = await apiClient.get(`/reviews/venue/${venueId}`, { params: { limit, offset } })
    return response.data
  },

  // دریافت خلاصه امتیازات یک سالن
  getVenueSummary: async (venueId: number): Promise<VenueRatingSummary> => {
    const response = await apiClient.get(`/reviews/venue/${venueId}/summary`)
    return response.data
  },

  // دریافت نظرات کاربر جاری
  getMyReviews: async (limit = 50): Promise<ReviewResponse[]> => {
    const response = await apiClient.get('/reviews/my', { params: { limit } })
    return response.data
  },

  // ثبت نظر جدید
  create: async (data: ReviewCreate): Promise<ReviewResponse> => {
    const response = await apiClient.post('/reviews/', data)
    return response.data
  },

  // ویرایش نظر
  update: async (reviewId: number, data: ReviewCreate): Promise<ReviewResponse> => {
    const response = await apiClient.put(`/reviews/${reviewId}`, data)
    return response.data
  },

  // حذف نظر
  delete: async (reviewId: number): Promise<void> => {
    await apiClient.delete(`/reviews/${reviewId}`)
  },
}

File: backend/app/schemas/review.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional

class ReviewCreate(BaseModel):
    venue_id: int
    rating: int = Field(..., ge=1, le=5, description="امتیاز ۱ تا ۵")
    comment: Optional[str] = Field(default=None, max_length=1000)

class ReviewResponse(BaseModel):
    id: int
    venue_id: int
    user_id: int
    rating: int
    comment: Optional[str]
    created_at: datetime

    # اطلاعات تکمیلی برای نمایش بهتر (نام کاربر و سالن)
    user_name: Optional[str] = None
    venue_name: Optional[str] = None

    class Config:
        from_attributes = True

class VenueRatingSummary(BaseModel):
    venue_id: int
    average_rating: float
    total_reviews: int
